test/constraint.jl

```diff
@@ -200,4 +200,55 @@
         @test macroexpand(:(@variable(m, -rand(5,5) <= nonsymmetric[1:5,1:5] <= rand(5,5), Symmetric))).head == :error
     end
 
+    @testset "[macros] sum(generator)" begin
+        m = Model()
+        @variable(m, x[1:3,1:3])
+        @variable(m, y)
+        C = [1 2 3; 4 5 6; 7 8 9]
+
+        cref = @constraint(m, sum( C[i,j]*x[i,j] for i in 1:2, j = 2:3 ) <= 1)
```

Comment:
It would be more direct to use `@expression` here so you don't need to re-extract the `AffExpr` from the constraint. These tests may have been written originally before `@expression` existed.